### PR TITLE
feat: [CDS-49054]: PDC dynamic provision

### PR DESCRIPTION
**Pull request by radujomir**
#### Screenshots

Screenshot 2022-12-27 at 21 30 56

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureInterface.tsx [hunk 1]
```diff
@@ -62,6 +70,20 @@ export const parseByComma = (data: string) =>
 
 export const parseHosts = (hosts: string) => parseByComma(hosts)
 
+export const getUpdateAttributes = (hostAttributes?: NGTag[]) => {
+  const result = {} as any
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureInterface.tsx:74]: Nit: `any` can be avoided here.

File: src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpec.tsx [hunk 2]
```diff
@@ -121,12 +129,15 @@ const PDCInfrastructureSpecEditable: React.FC<PDCInfrastructureSpecEditableProps
...
-  const [isPreconfiguredHosts, setIsPreconfiguredHosts] = useState(
-    initialValues.hosts ? PreconfiguredHosts.FALSE : PreconfiguredHosts.TRUE
+  const [selectionType, setSelectionType] = useState<HOSTS_TYPE>(
+    initialValues.isDynamicallyProvisioned
+      ? HOSTS_TYPE.DYNAMIC
+      : initialValues.hosts
+      ? HOSTS_TYPE.SPECIFIED
+      : HOSTS_TYPE.PRECONFIGURED
   )
+  const [showPreviewHostBtn, setShowPreviewHostBtn] = useState(selectionType !== HOSTS_TYPE.DYNAMIC)
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/PDCInfrastructureSpec/PDCInfrastructureSpec.tsx:140]: This state can be removed if `showPreviewHostBtn` can always be derived from `selectionType`.

**Comment by radujomir** [comment]
retrigger titlecheck

**Comment by radujomir** [comment]
retrigger coverage

**Review by litvinovstas — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Pull request merged**